service/node.go

```diff
@@ -504,9 +519,126 @@ func (s *service) NodeGetInfo(
 	}, nil
 }
 
-func (s *service) NodeGetVolumeStats(
-	ctx context.Context, req *csi.NodeGetVolumeStatsRequest) (*csi.NodeGetVolumeStatsResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "")
+// NodeGetVolumeStats will check the status of a volume given its ID and path
+// if volume is healthy, stats on volume usage will be returned
+// if volume is unhealthy, a message will be returned detailing the issue
+// To determine if volume is healthy, this method checks: volume known to array, volume known to SDC, volume path readable, and volume path mounted
+// Note: kubelet only calls this method when feature gate: CSIVolumeHealth=true
+func (s *service) NodeGetVolumeStats(ctx context.Context, req *csi.NodeGetVolumeStatsRequest) (*csi.NodeGetVolumeStatsResponse, error) {
+
+	csiVolID := req.GetVolumeId()
+	volPath := req.GetVolumePath()
+	mounted := false
+	healthy := true
+	message := ""
+
+	//validate params first, make sure neither field is empty
+	if len(volPath) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "no volume Path provided")
+	}
+
+	if len(csiVolID) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "no volume ID  provided")
+	}
+
+	//check if volume exists
+
+	volID := getVolumeIDFromCsiVolumeID(csiVolID)
+	systemID := s.getSystemIDFromCsiVolumeID(csiVolID)
+
+	if systemID == "" {
```

Comment:
Abhi's PR (#25) checks to see if a default system exists, do we want to do the same check here? Here is the code in question: https://github.com/dell/csi-powerflex/pull/25/files#diff-8120cde4a65d55a72e0ffe83dd202cadcc210138f8b1525e77ac20aa4e0c7388R2164

Reply:
added check